internal/provider/resource_packer_channel.go

```diff
@@ -0,0 +1,381 @@
+				Description:      "The slug of the HCP Packer Registry image bucket where the channel should be managed in.",
+				Type:             schema.TypeString,
+				ForceNew:         true,
+				Required:         true,
+				ValidateDiagFunc: validateStringNotEmpty,
+			},
+			// Optional inputs
+			"iteration_assignment": {
+				Description: "The iteration assignment information that will be used to assign a completed iteration to the channel.",
+				Type:        schema.TypeList,
+				MaxItems:    1,
+				Optional:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"fingerprint": {
+							ConflictsWith: []string{"iteration_assignment.0.id", "iteration_assignment.0.incremental_version"},
+							Description:   "The fingerprint of the iteration to assign to the channel.",
+							Type:          schema.TypeString,
+							Optional:      true,
+						},
+						"id": {
+							ConflictsWith: []string{"iteration_assignment.0.fingerprint", "iteration_assignment.0.incremental_version"},
+							Description:   "The id of the iteration to assign to the channel.",
+							Type:          schema.TypeString,
+							Optional:      true,
+						},
+						"incremental_version": {
+							ConflictsWith: []string{"iteration_assignment.0.fingerprint", "iteration_assignment.0.id"},
+							Description:   "The incremental_version of the iteration to assign to the channel.",
+							Type:          schema.TypeInt,
+							Optional:      true,
+							ValidateFunc:  validation.IntAtLeast(1),
+						},
+					},
+				},
+			},
+			// Computed Values
+			"author_id": {
+				Description: "The author of the channel.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"iteration": {
+				Description: "The iteration assigned to the channel.",
+				Type:        schema.TypeList,
+				Computed:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"fingerprint": {
+							Description: "The fingerprint of the iteration assigned to the channel.",
+							Type:        schema.TypeString,
+							Computed:    true,
+						},
+						"id": {
+							Description: "The id of the iteration assigned to the channel.",
+							Type:        schema.TypeString,
+							Computed:    true,
+						},
+						"incremental_version": {
+							Description: "The incremental_version of the iteration assigned to the channel.",
+							Type:        schema.TypeInt,
+							Computed:    true,
+						},
+					},
+				},
+			},
+			"organization_id": {
+				Description: "The ID of the organization this HCP Packer registry is located in.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"project_id": {
+				Description: "The ID of the project this HCP Packer registry is located in.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"created_at": {
+				Description: "Creation time of this build.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
```

Comment:
I think channels also have a `updated_at` time. Should it be added here as well?